storage/local/chunk.go

```diff
@@ -188,23 +188,27 @@ type chunk interface {
 	add(sample *metric.SamplePair) []chunk
 	clone() chunk
 	firstTime() clientmodel.Timestamp
-	lastTime() clientmodel.Timestamp
 	newIterator() chunkIterator
 	marshal(io.Writer) error
 	unmarshal(io.Reader) error
 	unmarshalFromBuf([]byte)
 	encoding() chunkEncoding
-	// values returns a channel, from which all sample values in the chunk
-	// can be received in order. The channel is closed after the last
-	// one. It is generally not safe to mutate the chunk while the channel
-	// is still open.
-	values() <-chan *metric.SamplePair
 }
 
 // A chunkIterator enables efficient access to the content of a chunk. It is
 // generally not safe to use a chunkIterator concurrently with or after chunk
 // mutation.
 type chunkIterator interface {
+	// length returns the number of samples in the chunk.
+	length() int
+	// Gets the timestamp of the n-th sample in the chunk.
+	getTimestampAtIndex(int) clientmodel.Timestamp
```

Comment:
Yeah, Go is kind of get... averse, but indeed, the rationale here is that it requires processing and is not really a classical OOP getter.

One might still think about removing the get's because Go really has no strict tradition of writing get... for things that require processing, either.

But that would be a change different to this one. I added new get... methods here only because the existing ones in the iterator were using that naming pattern.